src/utility/Ex3.py

```diff
@@ -0,0 +1,230 @@
+            continue
+        allEvents.append(sample)
+        i += 1
+
+    events = []
+
+    #Standardize impact of events and convert 6xn -> nx6 array
+    for i in range(0, n):
+        if(allEvents[0][i] < 0):
+            events.append([(-1)*abs(allEvents[j][i]) for j in range(0, emo_type_count) ])
+        else:
+            events.append([abs(allEvents[j][i]) for j in range(0, emo_type_count)])
+    return [events, n]
+
+def generate_ages(data) -> ([list, int]):
+    # n = data["ageDistribution"]["distribution"]["normal"]["samples"] # Using jsonref
+    n = data["numOfAgents"]["value"] #n = numOfAgents
+    k = data["ageDistribution"]["distribution"]["normal"]["numberOfValues"]
+    min_value = data["ageDistribution"]["distribution"]["normal"]["minValue"] # min_value = 5
+    max_value = data["ageDistribution"]["distribution"]["normal"]["maxValue"] # max_value = 104
+
+    mean = (min_value + max_value) / 2 # Mean
+    std = (max_value - min_value) / 6 # Standard deviation
+    while (True):
+        sample = np.random.normal(mean, std, n) #Samples follow normal distribution
+        sample = np.clip(sample, min_value, max_value) # Value trimming
+        sample = np.round(sample, 1)
+
+        stat, p_value = stats.kstest(sample, 'norm', args=(mean, std)) # Calc p_value
+
+        alpha = 0.05 # Level of significant
+        if p_value < alpha:
+            continue
+        else:
+            f = lambda x: ((x >= 23) & (x <= 61))
+            cond = f(sample)
+            sample = np.concatenate((np.extract(~cond, sample), np.extract(cond, sample)))
+            #Put special value of age first, prioritize Personels' age in this case
+            return [sample.tolist(), n]
+
+def generate_pedestrians(data):
+    pedestrians = []
+
+    num_agents = data["numOfAgents"]["value"]
+    num_noDisability = int(num_agents * (
+            data["walkability"]["distribution"]["noDisabilityNoOvertaking"]["value"] +
+            data["walkability"]["distribution"]["noDisabilityOvertaking"]["value"]) / 100)
+    num_personel = int(random.uniform(0.8, 0.9) * num_noDisability)
+
+    journeys = list(data["wardDistribution"]["distribution"].keys())[:-1]  # Exclude 'normal'
+    journey_distribution = data["walkability"]["distribution"]["journeyDistribution"]["distribution"]
+    for key in journey_distribution:
+        if key != "forPersonel" and journey_distribution[key]["start"] in journeys:
+            journeys.remove(journey_distribution[key]["start"])
+
+    num_openness = 0
+    num_neuroticism = 0
+
+    allEvents, eventCount = generate_events(data)
+    allTimeDistances, timeDistancesCount = generate_time(data)
+    allAge, ageCount = generate_ages(data) # Prioritize Personel first
+
+    while len(pedestrians) < num_agents:
+        age = allAge.pop() # Prioritize Personel first
+
+        if (num_neuroticism >= 0.53 * num_agents) | (age < 11): #Age < 11 must be "open" personality
+            personality = "open"
```

Comment:
`personality` is a class. Consider make a Personality class with 2 children: Open and Neurotic (used in Ex6)

Reply:
Or just map the string with a class in C++ when use and call it a day! Its values are constants anyway...